crates/credentials/src/verifiable_credential.rs

```diff
@@ -206,64 +252,87 @@ impl VerifiableCredential {
-        // check claims match expected values
-        if vc.id == String::default() {
-            vc.id = jti;
-        } else if vc.id != jti {
-            return Err(CredentialError::IdMismatch);
+        if let Some(id) = &vc_payload.id {
+            if id != &jti {
+                return Err(CredentialError::IdMismatch);
+            }
+        }
+
+        if let Some(issuer) = &vc_payload.issuer {
+            let vc_issuer = issuer.to_string();
+            if iss != vc_issuer {
+                return Err(CredentialError::IssuerMismatch);
+            }
         }
 
-        let vc_issuer = vc.issuer.to_string();
-        if vc_issuer.is_empty() {
-            vc.issuer = Issuer::String(iss.clone());
-        } else if iss != vc_issuer {
-            return Err(CredentialError::IssuerMismatch);
+        if let Some(credential_subject) = &vc_payload.credential_subject {
+            if sub != credential_subject.id {
+                return Err(CredentialError::SubjectMismatch);
+            }
         }
 
-        if vc.credential_subject.id == String::default() {
-            vc.credential_subject.id = sub;
-        } else if vc.credential_subject.id != sub {
-            return Err(CredentialError::SubjectMismatch);
+        if let Some(issuance_date) = &vc_payload.issuance_date {
+            if issuance_date != &nbf {
+                return Err(CredentialError::IssuanceDateMismatch);
+            }
         }
 
-        if vc.issuance_date == i64::default() {
-            vc.issuance_date = nbf;
-        } else if vc.issuance_date != nbf {
-            return Err(CredentialError::IssuanceDateMismatch);
+        if let Some(expiration_date) = &vc_payload.expiration_date {
+            if exp.is_none() {
+                return Err(CredentialError::MisconfiguredExpirationDate(
+                    "VC has expiration date but no exp in registered claims".to_string(),
+                ));
+            }
+
+            if expiration_date != &exp.unwrap() {
+                return Err(CredentialError::IssuanceDateMismatch);
+            }
         }
 
-        // if exp exists, make sure there is not a mismatch and assign it to vc expiration date
-        // if vc expiration date exists and exp does not, throw a misconfigured exp error
         if exp.is_some() {
-            if vc.expiration_date.is_some() {
-                if vc.expiration_date != exp {
-                    return Err(CredentialError::ExpirationDateMismatch);
-                }
-
-                let now = Utc::now().timestamp();
-                if now > exp.unwrap() {
-                    return Err(CredentialError::CredentialExpired(
-                        "The verifiable credential has expired".to_string(),
-                    ));
-                }
-
-                vc.expiration_date = exp;
+            let now = Utc::now().timestamp();
+            if now > exp.unwrap() {
```

Comment:
ditto https://github.com/TBD54566975/web5-rs/pull/217/files#r1609142601

Reply:
in this since we do `if exp.is_some()`, I'd prefer if we did `if let Some(exp) = exp`